src/FondOfSpryker/Zed/SizeSwitcher/Business/Publisher/ProductAbstractPageSearchPublisher.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace FondOfSpryker\Zed\SizeSwitcher\Business\Publisher;
+
+use FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToEventBehaviorFacadeInterface;
+use FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToProductPageSearchFacadeInterface;
+use FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToStoreFacadeInterface;
+use FondOfSpryker\Zed\SizeSwitcher\Persistence\SizeSwitcherRepositoryInterface;
+
+class ProductAbstractPageSearchPublisher implements ProductAbstractPageSearchPublisherInterface
+{
+    /**
+     * @var \FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToStoreFacadeInterface|\Generated\Shared\Transfer\StoreTransfer
+     */
+    protected $storeFacade;
+
+    /**
+     * @var \FondOfSpryker\Zed\SizeSwitcher\Persistence\SizeSwitcherRepositoryInterface
+     */
+    protected $sizeSwitcherRepository;
+
+    /**
+     * @var \FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToProductPageSearchFacadeInterface
+     */
+    protected $productPageSearchFacade;
+
+    /**
+     * @var \FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToEventBehaviorFacadeInterface
+     */
+    protected $eventBehaviorFacade;
+
+    /**
+     * @param \FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToStoreFacadeInterface $storeFacade
+     * @param \FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToEventBehaviorFacadeInterface $eventBehaviorFacade
+     * @param \FondOfSpryker\Zed\SizeSwitcher\Dependency\Facade\SizeSwitcherToProductPageSearchFacadeInterface $productPageSearchFacade
+     * @param \FondOfSpryker\Zed\SizeSwitcher\Persistence\SizeSwitcherRepositoryInterface $sizeSwitcherRepository
+     */
+    public function __construct(
+        SizeSwitcherToStoreFacadeInterface $storeFacade,
+        SizeSwitcherToEventBehaviorFacadeInterface $eventBehaviorFacade,
+        SizeSwitcherToProductPageSearchFacadeInterface $productPageSearchFacade,
+        SizeSwitcherRepositoryInterface $sizeSwitcherRepository
+    ) {
+        $this->storeFacade = $storeFacade;
+        $this->sizeSwitcherRepository = $sizeSwitcherRepository;
+        $this->productPageSearchFacade = $productPageSearchFacade;
+        $this->eventBehaviorFacade = $eventBehaviorFacade;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\EventEntityTransfer[] $eventTransfer
+     *
+     * @return void
+     */
+    public function publish(array $eventTransfer): void
+    {
+        $availabiltyIds = $this->getAvailabilityIdsFromEventTransfer($eventTransfer);
+
+        if ($availabiltyIds === null) {
+            return;
+        }
+
+        $availabiltyAbstractIds = $this->queryAbvailabiltyAbstractIdsByAvailabilityIds($availabiltyIds);
```

Comment:
Theoretisch kannst du dir den Schritt sparen oder? in spy_availability_abstract stehen ja auch die abstract_skus drin

Reply:
Das Update-Event hört auf spy_availability. Dann müsste ich prüfen ob die abstrake ID im EntityTransfer unter getForeignKeys() drin ist. Was ist besser?

Reply:
Dann habe ich mich durch die Namings in die Irre führen lassen und falsch geguckt. Kannst hier aber denke ich trotzdem ein paar Sachen sparen und im Repo mehr machen (joins, weitere query und mappings)